src/components/Divider.tsx

```diff
@@ -1,70 +1,113 @@
-import React, { useEffect } from 'react'
-import { useDispatch } from 'react-redux'
-import { useSelector } from 'react-redux'
+import { useCallback, useLayoutEffect, useRef, useState } from 'react'
+import { useDispatch, useSelector } from 'react-redux'
 import { css } from '../../styled-system/css'
 import { SystemStyleObject } from '../../styled-system/types'
 import Path from '../@types/Path'
+import State from '../@types/State'
+import ThoughtId from '../@types/ThoughtId'
 import { setCursorActionCreator as setCursor } from '../actions/setCursor'
-import { DIVIDER_MIN_WIDTH, DIVIDER_PLUS_PX } from '../constants'
+import { DIVIDER_MIN_WIDTH } from '../constants'
 import attributeEquals from '../selectors/attributeEquals'
+import { getAllChildrenAsThoughts } from '../selectors/getChildren'
 import rootedParentOf from '../selectors/rootedParentOf'
+import editingValueUntrimmedStore from '../stores/editingValueUntrimmed'
+import viewportStore from '../stores/viewport'
 import fastClick from '../util/fastClick'
 import head from '../util/head'
+import isDivider from '../util/isDivider'
+
+/** Custom hook to fetch relevant thought IDs for calculating divider width. */
+const useRelevantThoughtIds = (path: Path): ThoughtId[] => {
+  return useSelector(
+    (state: State) => {
+      const parentPath = rootedParentOf(state, path)
+      const parentId = head(parentPath)
+      const grandParentPath = parentId ? rootedParentOf(state, parentPath) : []
+      const grandParentId = head(grandParentPath)
+      const children = parentId ? getAllChildrenAsThoughts(state, parentId) : []
+      const childrenWithoutDividers = children.filter(child => !isDivider(child.value))
+      const isOnlyChild = childrenWithoutDividers.length === 0
+      const isTableView =
+        (parentId && attributeEquals(state, parentId, '=view', 'Table')) ||
+        (grandParentId && attributeEquals(state, grandParentId, '=view', 'Table'))
```

Comment:
You can make this a little more terse by omitting `parentId &&` since `attributeEquals` handles null.